loopout.go

```diff
@@ -706,7 +706,77 @@ func (s *loopOutSwap) waitForHtlcSpendConfirmed(globalCtx context.Context,
 		return nil, fmt.Errorf("register spend ntfn: %v", err)
 	}
 
+	var paymentComplete bool
+	pushPreimage := func(ctx context.Context) error {
+		// If our payment is complete, there is no need to proceed in
+		// attempting to push the preimage.
+		if paymentComplete {
+			return nil
+		}
+
+		// If we have not yet revealed our preimage through a sweep, we
+		// do not push the preimage because we may choose to never sweep
+		// if fees are too high.
+		if s.state != loopdb.StatePreimageRevealed {
+			return nil
+		}
+
+		// Create a child context for a one time lookup of our payment.
+		// We defer cancelling it so that lnd will terminate the stream.
+		trackCtx, cancel := context.WithCancel(ctx)
+		defer cancel()
+
+		// Lookup our payment in lnd before we push to the server, this
+		// allows us to exit early if the payment is already settled.
+		statusChan, errChan, err := s.lnd.Router.TrackPayment(
```

Comment:
Alternatively this call can be done once before we enter the republish loop. We just keep listening to the same status channel.